src/crafting.h

```diff
@@ -5,12 +5,14 @@
 #include "requirements.h" // requirement_data
 #include "cursesdef.h"    // WINDOW
 #include "string_id.h"
+#include "recipe_dictionary.h"
 
 #include <string>
 #include <vector>
 #include <map>
 #include <list>
 
+class recipe_dictionary;
```

Comment:
Either the include or the class declaration is necessary.  Preferably remove the include.